### ISSUE TITLE
Deprecation - warn users about watching the project directory

### ISSUE THREAD

**Issue by e2** [guard/guard#696]
### TL;DR - Guard should warn about watching the project directory instead of using the `directories` statement.

Reason: every time a file changes (e.g. ./log/_, ./db/sqlite._), the change goes through the whole stack (with some exceptions), meaning:
- the adapter backend (e.g. rb-fsevent - threads, system resources),
- Celluloid (threads, mutexes, thread/fiber switching)
- filesystem (dir entries, fstat, recursion)
- Listen::Record (hashes, mutexes, threads, Celluloid)
- Listen queue (wait, sleep, threads, locks, Celluloid)
- Listen silencer (Celluloid actor, regexps)
- callbacks (context switching)
- potentially killing guard interactor (to run tasks that actually match file, Pry methods)
- resetting the screen line
- getting screen attributes (e.g. notification, pry switching)
- Ruby callbacks
- symbol lookup (large symbol table in Guard when lots of plugins)
- heavy-processing of hashes with changes, lots of regexp matching, scope resolution and iteration in Guard
- lots of File.exist? with Dir.glob in plugins
- file relative/absolute conversion in Guard
- file changes processed twice in Guard (once to detect relevant changes, second to process them)
- and I've probably still missed a few things...

Listening to directories is the cheapest on Linux, but still...

In short, it's Apple's fault, because they made listening non-recursively impractical. Which means - wanting to watch a single file in the project root directory means automatically watching things like ./db/_sqlite, ./log/_.log, ./tmp, /vendor/, etc. - and every change goes through the stack above.

And because they made it impractical, to protect users from paying the price, users should be warned about watching the whole project directory - _especially_ on large or growing projects.

Also, even the reevaluating feature of Guard means forcing recursive watching of the whole project (so Guardfiles should be moved to config).

Ideally, Guard should allow "switching" which directories are watched based on the active scope (e.g. by collecting regexps and watching only directories with files it can respond to).
